saw-remote-api/python/tests/saw/test_provers.py

```diff
@@ -0,0 +1,43 @@
+from cryptol import cryptoltypes
+import saw
+from saw.proofscript import *
+
+import unittest
+from pathlib import Path
+
+
+def cry(exp):
+    return cryptoltypes.CryptolLiteral(exp)
+
+class ProverTest(unittest.TestCase):
+
+    @classmethod
+    def setUpClass(self):
+        saw.connect(reset_server=True)
+
+    @classmethod
+    def tearDownClass(self):
+        saw.reset_server()
+        saw.disconnect()
+
+    def test_provers(self):
+
+        if __name__ == "__main__": saw.view(saw.LogResults())
+
+        simple_thm = cry('\(x:[8]) -> x != x+1')
+        self.assertTrue(saw.prove(simple_thm, ProofScript([abc])).is_valid())
+        self.assertTrue(saw.prove(simple_thm, ProofScript([yices([])])).is_valid())
```

Comment:
These proof scripts look a little funny to me -- funny as in, a little low level and boiler-platey in the way they're constructed/specified, some of the solvers' names are class instances while others are functions that take a list of uninterpreted function names, etc... I guess this is probably fine as we're initially adding the feature, but I think we should try to come back and experiment with this API, look at ergonomics, name/usage consistency, etc  soon before we get a bunch of python proofs and have to worry about breaking things/refactoring/etc.

Reply:
I entirely agree.